bson/src/test/scala/org/mongodb/scala/bson/codecs/MacrosSpec.scala

```diff
@@ -39,6 +40,14 @@ class MacrosSpec extends FlatSpec with Matchers {
   case class DefaultValue(name: String, active: Boolean = false)
   case class SeqOfStrings(name: String, value: Seq[String])
   case class RecursiveSeq(name: String, value: Seq[RecursiveSeq])
+  case class Binary(binary: Array[Byte]) {
+    override def equals(arg: Any): Boolean = arg match {
+      case that: Binary => that.binary.deep == binary.deep
+      case _ => false
+    }
+  }
+  case class AllTheBsonTypes(documentMap: Map[String, String], array: Seq[String], date: Date, boolean: Boolean,
+                             double: Double, int32: Int, int64: Long, string: String, binary: Binary, none: Option[String])
```

Comment:
Well we need to test case classes and as `Array[Byte]` does equality based on the id of the object and not the value - it helps me cheat and override the equality testing for a much simpler case class.